### PR TITLE
MGMT-6712: Enable host validations for network latency and packet loss

### PR DESCRIPTION
**Pull request by jordigilh**
* Enables the host validations for network latency and packet loss with default values.
* The goal is to allow these 2 validations to be validated in staging and integration environments, so this PR addresses the changes required to allow system and unit tests to pass because of these 2 new validations.
* The validations will not be enabled in cloud deployments for now.  There is an MR downstream that disables them for production environments.
* Expose  the DISABLED_HOST_VALIDATIONS environment variable to for cloud deployments by adding it into the `template.yaml` file with the following configuration:
```
- name: DISABLED_HOST_VALIDATIONS
  value: ""
  required: false
``` 
* Expose DISABLED_HOST_VALIDATIONS environment variable for test deployments (assisted-test-infra and `make deploy-test` in the assisted-service project, for instance): It is now possible to inject the variable through the invoking shell.

### CODE REVIEW

File: data/default_hw_requirements.json [hunk 1]
```diff
@@ -4,13 +4,17 @@
...
+    "network_latency_threshold_ms": 100,
+    "packet_loss_percentage": 0
   },
   "worker": {
     "cpu_cores": 2,
     "ram_mib": 8192,
     "disk_size_gb": 120,
-    "installation_disk_speed_threshold_ms": 10
+    "installation_disk_speed_threshold_ms": 10,
+    "network_latency_threshold_ms": 500,
```
Comment (gamli75) [inline data/default_hw_requirements.json:16]: it's a different value than what we have in the saas.yaml https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data/services/assisted-installer/cicd/saas.yaml#L66
 Is it on purpose?

**Comment by ybettan** [comment]
@jordigilh What about the operator deployment? Is it supposed to use the default values? If not please make sure the operator sets the correct values https://github.com/openshift/assisted-service/blob/d10163ac70e0ccc10ec5de295db65a17b09978d8/internal/controller/controllers/agentserviceconfig_controller.go#L466-L526.

Everything else looks good to me exposing new env var wise @gamli75.

File: data/default_hw_requirements.json (hunk 1 above)
Comment (jordigilh) [reply data/default_hw_requirements.json:16]: Good catch. Should be the same. I'll change to 1000

**Comment by jordigilh** [comment]
> @jordigilh What about the operator deployment? Is it supposed to use the default values? If not please make sure the operator sets the correct values
> 
> https://github.com/openshift/assisted-service/blob/d10163ac70e0ccc10ec5de295db65a17b09978d8/internal/controller/controllers/agentserviceconfig_controller.go#L466-L526
> 
> .
> Everything else looks good to me exposing new env var wise @gamli75.

Thanks @ybettan for pointing it out. I missed that one :)

**Comment by jordigilh** [comment]
> /retest

you read my mind :smile:

**Comment by jordigilh** [comment]
@ybettan can you take a look at the PR and let me know if you have any further comments? Thanks :smile:

**Comment by gamli75** [comment]
@osherdp There is a problem with the commit message:  Your commit message should start with a JIRA issue ('JIRA-1111'), a GitHub issue ('#39'), or a BugZilla issue ('Bug 123')

[2021-06-09T13:46:12.336Z] with a following colon(:).

[2021-06-09T13:46:12.336Z] i.e. 'MGMT-42: Summary of the commit message'

I don't understand why there is a problem with the commit message?

File: Makefile [hunk 2]
```diff
@@ -265,7 +265,7 @@ deploy-service-requirements: | deploy-namespace deploy-inventory-service-file
 		--ocp-versions '$(subst ",\",$(OPENSHIFT_VERSIONS))' --public-registries "$(PUBLIC_CONTAINER_REGISTRIES)" \
 		--check-cvo $(CHECK_CLUSTER_VERSION) --apply-manifest $(APPLY_MANIFEST) $(ENABLE_KUBE_API_CMD) $(E2E_TESTS_CONFIG) \
 		--storage $(STORAGE) --ipv6-support $(IPV6_SUPPORT) --enable-sno-dnsmasq $(ENABLE_SINGLE_NODE_DNSMASQ) \
-		--hw-requirements '$(subst ",\",$(HW_REQUIREMENTS))' --kubeapi-day2 "$(ENABLE_KUBE_API_DAY2)"
+		--hw-requirements '$(subst ",\",$(HW_REQUIREMENTS))' --kubeapi-day2 "$(ENABLE_KUBE_API_DAY2)" --disabled-host-validations "$(DISABLED_HOST_VALIDATIONS)"
```
Comment (ybettan) [inline Makefile:268]: Can you please add it in a new line to keep lines short?

**Comment by ybettan** [comment]
I think you got all places now, I hope I didn't miss any.

The deployment process is so complex unfortunately :(

File: Makefile (hunk 2 above)
Comment (jordigilh) [reply Makefile:268]: Sure! :smile:

**Comment by jordigilh** [comment]
> I think you got all places now, I hope I didn't miss any.
> 
> The deployment process is so complex unfortunately :(

Agreed, it would be great to have it either documented or refactored for simplification. Or both :wink:

**Review by osherdp — APPROVED** [review]

**Pull request merged**